lib/pure/strutils.nim

```diff
@@ -2050,97 +2050,145 @@ func contains*(s: string, chars: set[char]): bool =
   ## * `find func<#find,string,set[char],Natural,int>`_
   return find(s, chars) >= 0
 
-func replace*(s, sub: string, by = ""): string {.rtl,
+func replace*(s: string, sub, by: char, maxOccurrence = -1): string {.rtl,
+    extern: "nsuReplaceChar".} =
+  ## Replaces `maxOccurrence` occurrences of the character `sub` in `s` with
+  ## the character `by`.
+  ## If `maxOccurrence` is set to `-1`, it replaces all occurrences.
+  ##
+  ## Optimized version of `replace <#replace,string,string,string>`_ for
+  ## characters.
+  ##
+  ## See also:
+  ## * `find func<#find,string,char,Natural,int>`_
+  ## * `replaceWord func<#replaceWord,string,string,string>`_
+  ## * `multiReplace func<#multiReplace,string,varargs[]>`_
+  runnableExamples:
+    doAssert "valid variable name".replace(' ', '_') == "valid_variable_name"
```

Comment:
* use assert in runnableExamples (and doAssert in all other tests), see contributing.html

Reply:
In many places `doAssert` was already used in `runnableExamples`, which weren't made by me.
Then it would make sense to gave some PR's that change all...?

Reply:
new code should stick to the currently accepted best practice, not surrounding code; and yes, a PR that changes all those should be done (and in fact i wrote it here: https://github.com/nim-lang/Nim/pull/17616 but i still need to undraft it even though i think it's ready)